pkg/operator/controllers/muo/muo_controller_test.go

```diff
@@ -0,0 +1,163 @@
+			enabled:             true,
+			managed:             "false",
+			expectDelete:        true,
+			returnErrorOnEnsure: true,
+			wantErr:             true,
+			errorString:         "Error removing MUO:\nfailed ensureDelete",
+		},
+		{
+			name:    "managed=blank (no action)",
+			enabled: true,
+			managed: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			cluster := &arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					OperatorFlags: arov1alpha1.OperatorFlags{
+						enabled:  strconv.FormatBool(tt.enabled),
+						pullSpec: "wonderfulPullspec",
+					},
+				},
+			}
+			// nil and empty string are valid values, be careful to preserve them
+			if tt.managed != "" {
+				cluster.Spec.OperatorFlags[managed] = tt.managed
+			}
+			arocli := arofake.NewSimpleClientset(cluster)
+			k8scli := fake.NewSimpleClientset()
+			dh := mock_dynamichelper.NewMockInterface(controller)
+
+			// mock the dynamichelper's Ensure
+			if tt.expectCreate && !tt.returnErrorOnEnsure {
+				ensure := func(ctx context.Context, objs ...kruntime.Object) error {
+					for _, i := range objs {
+						// When we "ensure" the object, apply the deployment
+						if d, ok := i.(*appsv1.Deployment); ok {
+							d.Status.AvailableReplicas = int32(tt.replicas)
+							d.Status.UpdatedReplicas = int32(tt.replicas)
+							d.Status.Replicas = 1
+							_, err := k8scli.AppsV1().Deployments(d.ObjectMeta.Namespace).Create(ctx, d, metav1.CreateOptions{})
+							if err != nil {
+								t.Fatal(err)
+							}
+						}
+					}
+					return nil
+				}
+				dh.EXPECT().Ensure(gomock.Any(), gomock.Any()).Do(ensure).Return(nil)
+			} else if tt.expectCreate && tt.returnErrorOnEnsure {
+				dh.EXPECT().Ensure(gomock.Any(), gomock.Any()).Return(errors.New("failed ensure"))
+			}
+
+			// mock the dynamichelper's EnsureDeleted
+			if tt.expectDelete && !tt.returnErrorOnEnsure {
+				dh.EXPECT().EnsureDeleted(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).MinTimes(1).Return(nil)
+			} else if tt.expectDelete && tt.returnErrorOnEnsure {
+				dh.EXPECT().EnsureDeleted(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).MinTimes(1).Return(errors.New("failed ensureDelete"))
+			}
+
+			r := &Reconciler{
+				arocli:            arocli,
+				kubernetescli:     k8scli,
+				dh:                dh,
+				deployer:          newDeployer(k8scli, dh),
+				readinessTimeout:  0 * time.Second,
+				readinessPollTime: 1 * time.Second,
+			}
+			_, err := r.Reconcile(context.Background(), reconcile.Request{})
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Reconcile() error = %v, wantErr = %v", err, tt.wantErr)
+				return
+			} else if tt.wantErr && err.Error() != tt.errorString {
+				t.Errorf("Reconcile() error string = '%v', errorString = '%v'", err, tt.errorString)
+			}
```

Comment:
I did something a bit more consistent. FWIW I extremely dislike the existing practice because it makes knowing what went on when it's not your machine (e.g. CI) harder.